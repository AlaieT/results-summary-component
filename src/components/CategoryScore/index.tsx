import React from "react";

import type { CategoryScoreProps } from "../../types";

import { ReactComponent as IconReaction } from "../../assets/images/icon-reaction.svg";
import { ReactComponent as IconMemory } from "../../assets/images/icon-memory.svg";
import { ReactComponent as IconVerbal } from "../../assets/images/icon-verbal.svg";
import { ReactComponent as IconVisual } from "../../assets/images/icon-visual.svg";

import styles from "../../styles/components/categoryScore.module.scss";

const ICONS = {
  Reaction: <IconReaction width={24} height={24} />,
  Memory: <IconMemory width={24} height={24} />,
  Verbal: <IconVerbal width={24} height={24} />,
  Visual: <IconVisual width={24} height={24} />,
};

const CategoryScore = ({ category, score }: CategoryScoreProps) => {
  return (
    <div className={styles.container + " " + styles[category]}>
      <div className={styles.category}>
        {ICONS[category]}
        {category}
      </div>
      <div className={styles.score}>
        {score}
        <span>&nbsp;/&nbsp;100</span>
      </div>
    </div>
  );
};

export default CategoryScore;
