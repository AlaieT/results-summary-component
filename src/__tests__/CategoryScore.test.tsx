import React from "react";
import { render, cleanup, screen } from "@testing-library/react";

import CategoryScore from "../components/CategoryScore";

afterEach(cleanup);

describe("CategoryScore", () => {
  describe("correct render", () => {
    it("should match snapshot", () => {
      const { asFragment } = render(
        <CategoryScore category="Memory" score={50} />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it("should display correct score", async () => {
      render(<CategoryScore category="Memory" score={50} />);

      expect(await screen.findByText("50")).toBeTruthy();
    });
  });
});
