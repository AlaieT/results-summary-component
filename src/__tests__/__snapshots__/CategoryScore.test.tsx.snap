// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryScore correct render should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container Memory"
  >
    <div
      class="category"
    >
      <div
        height="24"
        width="24"
      />
      Memory
    </div>
    <div
      class="score"
    >
      50
      <span>
         / 100
      </span>
    </div>
  </div>
</DocumentFragment>
`;
